### PR TITLE
feat: enable vulnerability assessment email notifications

### LINKED ISSUE #67: chore: should mssql vuln assessment email notifications be enabled (SNYK-CC-AZURE-540)
SNYK-issue for the `azurerm_mssql_server_vulnerability_assessment`-resource: 
- MSSQL Vulnerability Assessment email notifications are disabled ([SNYK-CC-AZURE-540](https://snyk.io/security-rules/SNYK-CC-AZURE-540))
  - **Severity:** `Low`
  - **Issue:** MSSQL Vulnerability Assessment email notifications are disabled
  - **Impact:** Administrators will not be automatically notified, which can lead to time delay in identifying risks and taking corrective measures
  - **Resolve:** Set `recurring_scans.email_subscription_admins` to `true`

### PR DESCRIPTION
**Pull request by kov117**
Closes #67

### CODE REVIEW

**Comment by kov117** [comment]
Test passed 👍🏽, changes has been made in the resource "azurerm_mssql_server_vulnerability_assessment" "this", recurring scans now has variables. This was done for the reason of not hardcoding in recurring scans, thus making it easier to make changes if need be.

File: main.tf [hunk 1]
```diff
@@ -152,8 +152,8 @@ resource "azurerm_mssql_server_vulnerability_assessment" "this" {
   storage_account_access_key      = var.storage_account_access_key
 
   recurring_scans {
-    enabled                   = true
-    email_subscription_admins = false
-    emails                    = []
+    enabled                   = var.vulnerability_assessment_recurring_scan_enabled
+    email_subscription_admins = var.vulnerability_assessment_recurring_scan_email_subscription_admins
+    emails                    = var.vulnerability_assessment_recurring_scan_emails
```
Comment (hknutsen) [inline main.tf]: Rename variables from `*_recurring_scan_*` to `*_recurring_scans_*`:

```suggestion
    enabled                   = var.vulnerability_assessment_recurring_scans_enabled
    email_subscription_admins = var.vulnerability_assessment_recurring_scans_email_subscription_admins
    emails                    = var.vulnerability_assessment_recurring_scans_emails
```

To follow variable naming convention = resource name + block name + argument name.

File: variables.tf [hunk 2]
```diff
@@ -121,6 +121,24 @@ variable "storage_container_name" {
...
+  type        = bool
+  default     = true
+}
+
+variable "vulnerability_assessment_recurring_scan_emails" {
+  description = "Specifies an array of email addresses to which the scan notification."
+  type        = list(string)
+  default     = []
+}
+
```
Comment (hknutsen) [inline variables.tf]: Rename variables from `*_recurring_scan_*` to `*_recurring_scans_*`:

```suggestion
variable "vulnerability_assessment_recurring_scans_enabled" {
  description = "Specifies if recurring scans is enabled or disabled."
  type        = bool
  default     = true
}

variable "vulnerability_assessment_recurring_scans_email_subscription_admins" {
  description = "Specifies if the schedule scan notification will be sent to the subscriptions administrators for this SQL server."
  type        = bool
  default     = true
}

variable "vulnerability_assessment_recurring_scans_emails" {
  description = "Specifies an array of email addresses to which the scan notification."
  type        = list(string)
  default     = []
}
```

**Review by hknutsen — CHANGES_REQUESTED** [review]
Nitpicking, but suggesting to rename variables 🙂

**Comment by hknutsen** [comment]
@kov117 I'd recommend renaming the PR to something like `feat: enable vulnerability assessment email notifications`.

Always think of it from the POV of the user. My suggestion would translate to something like `This feature allows you to enable vulnerability assessment email notifications`.

**Review by hknutsen — APPROVED** [review]
Looks good to me, and tests pass 🎉

**Pull request merged**